src/main/java/org/commcare/formplayer/services/EntitiesSelectionService.java

```diff
@@ -0,0 +1,75 @@
+package org.commcare.formplayer.services;
+
+import com.timgroup.statsd.StatsDClient;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.commcare.core.interfaces.EntitiesSelectionCache;
+import org.commcare.formplayer.objects.EntitiesSelection;
+import org.commcare.formplayer.repo.EntitiesSelectionRepo;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cache.Cache;
+import org.springframework.cache.CacheManager;
+import org.springframework.cache.annotation.CacheConfig;
+import org.springframework.cache.annotation.CacheEvict;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.stereotype.Service;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Optional;
+
+import javax.annotation.Nullable;
+
+@Service
+@CacheConfig(cacheNames = {"entities_selection"})
```

Comment:
@snopoke I am not quite sure if we actually need a cache on top of the postgres table. Do you have an opinion on it ?

Reply:
I think it makes sense to cache it for the sake of faster access.